src/commands/admin_commands.py

```diff
@@ -0,0 +1,128 @@
+import os
+import logging
+from discord import app_commands, Interaction
+
+# Retrieve the list of admin usernames from the .env file
+ADMIN_USERS = os.getenv("ADMIN_USERS", "").split(",")
+
+
+class AdminCommands(app_commands.Group):
+    def __init__(self, gemini_bot):
+        super().__init__(name="admin", description="Admin commands for DuckBot setup.")
+
+        # Add subgroups to the main admin group
+        self.set = SetSubGroup(self.check_admin)
+        self.reset = ResetSubGroup(self.check_admin, gemini_bot)
+
+        # Register subgroups
+        self.add_command(self.set)
+        self.add_command(self.reset)
+
+    async def check_admin(self, interaction: Interaction) -> bool:
+        user_name = interaction.user.name
+        logging.info(f"Checking admin status for user: {user_name}")
+
+        if user_name in ADMIN_USERS:
+            logging.info(f"User {user_name} is authorized.")
+            return True
+        else:
+            logging.warning(f"User {user_name} is not authorized.")
+            return False
+
+    async def interaction_check(self, interaction: Interaction) -> bool:
+        """Restrict all admin commands visibility to authorized users only."""
+        is_admin = await self.check_admin(interaction)
+        if is_admin:
+            return True
+        await interaction.response.send_message("Unauthorized", ephemeral=True)
+        return False
```

Comment:
This function doesn't seem to be currently used and only `check_admin` is used. You can probably combine these functions so that `check_admin` prints the unauthorised message if needed.